Makefile

```diff
@@ -49,4 +49,4 @@ $(opam_file): dune-project ## Update the package dependencies when new deps are
 
 init: ## Create a local opam switch and setups githooks
 	git config core.hooksPath .githooks
-	opam switch create . 4.10.0 --deps-only --with-test
+	opam switch create . --deps-only --with-test
```

Comment:
Why not updating version here?

Reply:
It just seems unnecessary. I also think not specifying it as an invariant might make upgrading easier, since then you shouldn't need to reinstall the whole switch. But I haven't actually tried that.

Reply:
I was thinking if there is no version explicitly mentioned in `make init` command, one would install 4.14 (or whatever is latest) just to switch back to 4.12 or whatever we are using for development 🤔

Reply:
Actually, on second thought, what would happen is that we'd develop with latest version of the compiler, while ci is tied to 4.12. Maybe we should leave it like that (to potentially detect issues in development asap), and enable multi-version ci in the future.

In any case, none of this is blocker for this PR.

Reply:
It actually installed 4.13.1 for me, for some reason. We could probably add a constraint on `dev` or `with-test` to have more precise control over this as well.